keps/184-scheduler-simulation/README.md

```diff
@@ -0,0 +1,214 @@
+
+But, we cannot define "which scheduler will do it" in `Scenario`. 
+It can't be helped because only one scheduler is running in a simulator and `Scenario` resource is created in a simulator.
+
+Wouldn't it be nice to be able to run the same scenario with various schedulers and see which scheduler is the best one? Yes. That's one of the motivations. 
+
+### Goals
+
+- In `SchedulerSimulation` resource, users can define the `Scenario` and the `Simulator`'s spec which run that `Scenario`. 
+- `SchedulerSimulation` resources (precisely the controller for `SchedulerSimulation`) will create the `Simulator` resource to run a simulator and run the `Scenario` in that simulator.
+
+### Non-Goals
+
+Create `SchedulerSimulation` in Web UI. (may be implemented in the future, but out of scope of this proposal.)
+
+## Proposal
+
+### Implementation design details
+
+#### Required prior knowledge
+
+The `Scenario` is the CRD to run scenario-based simulation introduced in KEP-140.
+It is applied to kube-apiserver started in kube-scheduler-simulator, (all simulator has own kube-apiserver internally.)
+and a scenario is start to running just after created.
+
+The `Simulator` is the CRD to run the simulator as Pod in the cluster introduced in KEP-167.
+It is applied to user's cluster (not to kube-apiserver started in a simulator.)
+
+#### SchedulerSimulation CRD
+
+The CRD `SchedulerSimulation` will be applied to kube-apiserver in user's cluster. (not to simulator's kube-apiserver.) 
+
+It's namespaced resource. 
+
+```go
+type SchedulerSimulation struct {
+  metav1.TypeMeta 
+  metav1.ObjectMeta // namespaced.
+
+  Spec SchedulerSimulationSpec
+  Status SchedulerSimulationStatus
+}
+  
+type SchedulerSimulationSpec struct {
+  // Describes the simulator that will be created.
+  // Cannot be updated.
+  SimulatorSpec SimulatorSpec
+  // ScenarioTemplateFilePath represents the path for file that has the Scenario resource's definition.
+  // The Scenario defined in the file will be run in the simulator.
+  //
+  // It should be the path for the mounted path in most cases,
+  // otherwise, the scenario won't created successfully.
+  //
+  // Cannot be updated.
+  ScenarioTemplateFilePath string 
+  // ScenarioResultFilePath represents the file path that the result of scenario will be stored.
+  // 
+  // It should be the path for the mounted path in most cases,
+  // otherwise, the simulation result won't stored to the persistent file.
+  //
+  // Cannot be updated.
+  ScenarioResultFilePath string
+  // Volume to mount into the filesystem.
+  //
+  // The container named "scenario-runner" will be created in the simulator Pod.
+  // That container will see the scenario defined in ScenarioTemplateFilePath, create the scenario in the simulator, and store the result into ScenarioReusltFilePath.
+  // The volume specified in Volume and VolumeMount is mounted in that container.
+  //
+  // Cannot be updated.
+  VolumeMount corev1.VolumeMount 
```

Comment:
Do we need to allow users to configure the path/volumeMount for them?? I suppose the fixed path/volumeMount would be enough, i.e., this raises any inconvenience for them. WDYT?

Reply:
> I suppose the fixed path/volumeMount would be enough, i.e., this raises any inconvenience for them. WDYT?

I may not have understood exactly what you meant. So, you wanna remove `VolumeMount` and make `Volume` always be mounted in the same `mountPath` in "scenario-runner" container?

Reply:
make sense.